### DISCUSSION TITLE
Watchdog callback

### DISCUSSION THREAD

**Discussion by LeeTroy** [DMTF/libspdm#2308] [General]
Hi,

In `os_stub/platform_lib/watchdog.c` provide the watchdog apis for start, stop and reset.
What should be done once the watchdog expires? Deleting the session with session_id?

https://github.com/DMTF/libspdm/blob/main/os_stub/platform_lib/watchdog.c

Thanks,
Troy Lee

**Comment by steven-bellock** [comment]
>Deleting the session with session_id?

Something like that, however there are currently some issues. Within libspdm
```
    libspdm_secured_message_set_session_state(
        session_info->secured_message_context,
        LIBSPDM_SESSION_STATE_NOT_STARTED);
    libspdm_free_session_id(spdm_context, session_id);
```    
is used to terminate the session. However both of those functions are internal to libspdm. Ideally the Integrator would just call one public function to terminate the session. In addition `libspdm_free_session_id` does more than just free the session ID, it also clears the secured message context, and should be renamed.

**Reply by LeeTroy** [reply]
Thanks!

**Comment by jyao1** [comment]
See https://github.com/DMTF/libspdm/issues/2309
